packages/Popover/src/Popover.js

```diff
@@ -0,0 +1,500 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { emotion, ccClassName } from '@leafygreen-ui/lib';
+
+const { css } = emotion;
+
+const rootPopoverStyle = css`
+  transition: transform 150ms ease-in-out, opacity 150ms ease-in-out;
+  position: absolute;
+  top: 0;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+export default class Popover extends Component {
+  static displayName = 'Popover';
+
+  static propTypes = {
+    children: PropTypes.node,
+    isActive: PropTypes.bool,
+    className: PropTypes.string,
+    align: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
+    justify: PropTypes.oneOf(['start', 'middle', 'end']),
+    refEl: PropTypes.any,
+    withoutPortal: PropTypes.bool,
+    getUpdatePosition: PropTypes.func,
+  };
+
+  static defaultProps = {
+    align: 'top',
+    justify: 'start',
+  };
+
+  // I know you don't normally like constructors, but it is a clean way to group state/ref declaration
+  // Otherwise wasn't sure where to put the ref declaration in a pattern-y way
+  constructor() {
+    super();
+
+    this.state = {
+      windowHeight: window.innerHeight,
+      windowWidth: window.innerWidth,
+      hasMounted: false,
+    };
+
+    this.contentRef = React.createRef();
+  }
+
+  componentDidMount() {
+    const { getUpdatePosition } = this.props;
+
+    if (getUpdatePosition) {
+      getUpdatePosition(this.updatePosition.bind(this, true));
+    }
+
+    const referenceElPos = this.getRefPosition(this.props.refEl);
+
+    this.setState({ hasMounted: true, referenceElPos });
```

Comment:
I know it's an artificial way but I was doing a lot of research into how to handle this, and I'll find the ticket but the React team suggested this approach in one instance. Definitely open to alternatives though. 

As far as `referenceElPos` goes, this is here because I don't want to call this.getRefPosition in all of the other methods, so I'm just calling it once, here, and leveraging it all around the class methods.